File: src/app/shared/service/api/unit.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpParams } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from 'src/environments/environment';
import { UnitDTORequest } from '../../model/request/unitDTORequest';

@Injectable({
  providedIn: 'root'
})

export class UnitService {

  constructor(
    private _httpClient: HttpClient
  ) { }

  findAll(key_word: string, page: number, size: number): Observable<any> {
    let queryParameters = new HttpParams();
    queryParameters = queryParameters.append('page', page);
    queryParameters = queryParameters.append('size', size);
    queryParameters = queryParameters.append('field', '');
    queryParameters = queryParameters.append('order', '');
    queryParameters = queryParameters.append('key_word', key_word);
    return this._httpClient.get<any>(environment.urlBase + '/units/unit?' + queryParameters);
  }

  public save(unit: UnitDTORequest): Observable<any> {
    return this._httpClient.post<any>(environment.urlBase + '/units/unit', unit);
  }

  public findById(id: number): Observable<any> {
    return this._httpClient.get<any>(environment.urlBase + '/units/unit/' + id);
  }

  public update(unit: UnitDTORequest, id: number): Observable<any> {
    return this._httpClient.put<any>(environment.urlBase + '/units/unit/' + id, unit);
  }

  public delete(id: number): Observable<any> {
    return this._httpClient.patch<any>(environment.urlBase + '/units/unit/' + id, null);
  }
}

File: src/main/java/com/sys/botica/crce/pe/sys_botica/service/UnitService.java
package com.sys.botica.crce.pe.sys_botica.service;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import com.sys.botica.crce.pe.sys_botica.dto.HrefEntityDTO;
import com.sys.botica.crce.pe.sys_botica.dto.UnitDTO;
import com.sys.botica.crce.pe.sys_botica.dto.request.UnitDTORequest;

public interface UnitService {

	public Page<UnitDTO> findByDescription(String key_word,Pageable pageable);
	public HrefEntityDTO save(UnitDTORequest unit);
	public HrefEntityDTO update(UnitDTORequest unit,Integer id);
	public HrefEntityDTO delete(Integer id);
	public UnitDTO findById(Integer id);
}
